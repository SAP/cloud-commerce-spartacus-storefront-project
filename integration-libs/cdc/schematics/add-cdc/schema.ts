import { LibraryOptions } from '@spartacus/schematics';

export interface Schema extends LibraryOptions {
  baseSite?: string;
  jsSDKUrl?: string;
  sessionExpiration?: number;
}
