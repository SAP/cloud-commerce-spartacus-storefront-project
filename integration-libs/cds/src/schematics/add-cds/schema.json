{
  "$schema": "http://json-schema.org/schema",
  "id": "CDSSchematics",
  "title": "CDS Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "tenant": {
      "type": "string",
      "description": "The tenant name.",
      "default": "my-tenant",
      "x-prompt": "What is your tenant name?"
    },
    "baseUrl": {
      "type": "string",
      "description": "The base URL.",
      "default": "https://my-tenant.api.us.context.cloud.sap",
      "x-prompt": "What is your base URL?"
    },
    "profileTagLoadUrl": {
      "type": "string",
      "description": "Where to load ProfileTag from. (e.g. https://tag.static.us.context.cloud.sap/js/profile-tag.js)",
      "x-prompt": "Where to load ProfileTag from? (optional)"
    },
    "profileTagConfigUrl": {
      "type": "string",
      "description": " Configuration URL for ProfileTag. (e.g. https://tag.static.stage.context.cloud.sap/config/my-config123)",
      "x-prompt": "What is the ProfileTag configuration URL? (optional)"
    }
  },
  "required": ["tenant", "baseUrl"]
}
