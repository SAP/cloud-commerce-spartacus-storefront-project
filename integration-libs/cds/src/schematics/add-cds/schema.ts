import { LibraryOptions } from '@spartacus/schematics';

export interface Schema extends LibraryOptions {
  tenant: string;
  baseUrl: string;
  strategyId: string;
  profileTagLoadUrl?: string;
  profileTagConfigUrl?: string;
}
