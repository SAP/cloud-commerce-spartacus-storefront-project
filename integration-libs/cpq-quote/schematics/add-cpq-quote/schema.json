{
  "$schema": "http://json-schema.org/schema",
  "$id": "CpqQuoteSchematics",
  "title": "cpq-quote Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display Cpq Discount Percentage",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the cpq-quote.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["cpq-quote"]
    }
  },
  "required": []
}
