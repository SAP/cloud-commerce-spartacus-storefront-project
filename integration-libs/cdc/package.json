{
  "name": "@spartacus/cdc",
  "version": "0.302.0-next.2",
  "description": "Customer Data Cloud Integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "customer data cloud",
    "gigya",
    "cdc"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/cdc",
  "license": "Apache-2.0",
  "dependencies": {
    "tslib": "^2.0.0"
  },
  "peerDependencies": {
    "@angular/common": "^10.1.0",
    "@angular/core": "^10.1.0",
    "@angular/router": "^10.1.0",
    "@ngrx/effects": "^10.0.0",
    "@ngrx/store": "^10.0.0",
    "@spartacus/core": "3.2.0-next.2",
    "@spartacus/storefront": "3.2.0-next.2",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
