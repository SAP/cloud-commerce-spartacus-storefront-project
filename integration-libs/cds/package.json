{
  "name": "@spartacus/cds",
  "version": "3.2.0-next.1",
  "description": "Context Driven Service library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "merchandising",
    "segmentation",
    "context-driven services",
    "cds"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/cds",
  "license": "Apache-2.0",
  "dependencies": {
    "tslib": "^2.0.0"
  },
  "peerDependencies": {
    "@angular/common": "^10.1.0",
    "@angular/core": "^10.1.0",
    "@angular/router": "^10.1.0",
    "@ngrx/store": "^10.0.0",
    "@spartacus/core": "3.2.0-next.1",
    "@spartacus/storefront": "3.2.0-next.1",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
