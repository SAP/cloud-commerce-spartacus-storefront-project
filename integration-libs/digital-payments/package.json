{
  "name": "@spartacus/digital-payments",
  "version": "6.7.0",
  "description": "Digital Payments Integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "digital payments",
    "payments"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/digital-payments",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^16.2.7",
    "@angular/common": "^16.2.10",
    "@angular/core": "^16.2.10",
    "@angular/forms": "^16.2.10",
    "@angular/router": "^16.2.10",
    "@ng-select/ng-select": "^11.1.1",
    "@spartacus/cart": "6.6.0-1",
    "@spartacus/checkout": "6.6.0-1",
    "@spartacus/core": "6.6.0-1",
    "@spartacus/schematics": "6.6.0-1",
    "@spartacus/storefront": "6.6.0-1",
    "rxjs": "^7.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
