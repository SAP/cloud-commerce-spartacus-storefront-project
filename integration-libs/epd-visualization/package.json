{
  "name": "@spartacus/epd-visualization",
  "version": "2210.70.0",
  "description": "SAP Enterprise Product Development Visualization integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "epd-visualization",
    "2d",
    "3d",
    "visual",
    "spare",
    "part",
    "picking",
    "epd"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/epd-visualization",
  "license": "Apache-2.0",
  "exports": {
    ".": {
      "sass": "./_index.scss"
    }
  },
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^17.0.5",
    "@angular/common": "^17.0.5",
    "@angular/core": "^17.0.5",
    "@angular/forms": "^17.0.5",
    "@angular/router": "^17.0.5",
    "@sapui5/ts-types-esm": "1.120.1",
    "@spartacus/cart": "2210.70.0",
    "@spartacus/core": "2210.70.0",
    "@spartacus/schematics": "2210.70.0",
    "@spartacus/storefront": "2210.70.0",
    "@spartacus/styles": "2210.70.0",
    "bootstrap": "^4.6.2",
    "rxjs": "^7.8.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
