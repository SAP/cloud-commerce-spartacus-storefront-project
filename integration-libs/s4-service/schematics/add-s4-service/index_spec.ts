/// <reference types="jest" />

import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import {
  S4_SERVICE_FEATURE_NAME,
  SPARTACUS_SCHEMATICS,
  SPARTACUS_S4_SERVICE,
  LibraryOptions as S4ServiceOptions,
  SpartacusOptions,
  s4ServiceFeatureModulePath,
} from '@spartacus/schematics';
import * as path from 'path';
import { peerDependencies } from '../../package.json';

const collectionPath = path.join(__dirname, '../collection.json');

describe('Spartacus S/4HANA Service Integration Schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner(
    SPARTACUS_S4_SERVICE,
    collectionPath
  );

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
    standalone: false,
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const libraryNoFeaturesOptions: S4ServiceOptions = {
    project: 'schematics-test',
    lazy: false,
    features: [],
  };

  const s4ServiceFeatureOptions: S4ServiceOptions = {
    ...libraryNoFeaturesOptions,
    features: [S4_SERVICE_FEATURE_NAME],
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      SPARTACUS_SCHEMATICS,
      path.join(
        __dirname,
        '../../../../projects/schematics/src/collection.json'
      )
    );

    appTree = await schematicRunner.runExternalSchematic(
      '@schematics/angular',
      'workspace',
      workspaceOptions
    );

    appTree = await schematicRunner.runExternalSchematic(
      '@schematics/angular',
      'application',
      appOptions,
      appTree
    );

    appTree = await schematicRunner.runExternalSchematic(
      SPARTACUS_SCHEMATICS,
      'ng-add',
      { ...spartacusDefaultOptions, name: 'schematics-test' },
      appTree
    );
  });

  describe('Without features', () => {
    beforeEach(async () => {
      appTree = await schematicRunner.runSchematic(
        'ng-add',
        libraryNoFeaturesOptions,
        appTree
      );
    });

    it('should not create any of the feature modules', () => {
      expect(appTree.exists(s4ServiceFeatureModulePath)).toBeFalsy();
    });
  });

  describe('S/4HANA Service Integration feature', () => {
    describe('general setup', () => {
      beforeEach(async () => {
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          {
            ...s4ServiceFeatureOptions,
          },
          appTree
        );
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          s4ServiceFeatureOptions,
          appTree
        );
      });

      it('should install necessary Spartacus libraries', () => {
        const packageJson = JSON.parse(appTree.readContent('package.json'));
        let dependencies: Record<string, string> = {};
        dependencies = { ...packageJson.dependencies };
        dependencies = { ...dependencies, ...packageJson.devDependencies };

        for (const toAdd in peerDependencies) {
          if (
            !peerDependencies.hasOwnProperty(toAdd) ||
            toAdd === SPARTACUS_SCHEMATICS
          ) {
            continue;
          }
          const expectedDependency = dependencies[toAdd];
          expect(expectedDependency).toBeTruthy();
        }
      });

      it('should add the feature using the lazy loading syntax', async () => {
        const module = appTree.readContent(s4ServiceFeatureModulePath);
        expect(module).toMatchSnapshot();
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          {
            ...s4ServiceFeatureOptions,
          },
          appTree
        );
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          { ...s4ServiceFeatureOptions, lazy: false },
          appTree
        );
      });

      it('should import appropriate modules', async () => {
        const module = appTree.readContent(s4ServiceFeatureModulePath);
        expect(module).toMatchSnapshot();
      });
    });
  });
});
