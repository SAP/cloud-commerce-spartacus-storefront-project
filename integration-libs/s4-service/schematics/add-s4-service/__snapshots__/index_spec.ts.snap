// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus S/4HANA Service Integration Schematics: ng-add S/4HANA Service Integration feature eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { I18nConfig, provideConfig } from "@spartacus/core";
import { S4ServiceModule } from "@spartacus/s4-service";
import { s4ServiceTranslationChunksConfig, s4ServiceTranslations } from "@spartacus/s4-service/assets";
import { S4ServiceRootModule } from "@spartacus/s4-service/root";

@NgModule({
  declarations: [],
  imports: [
    S4ServiceRootModule,
    S4ServiceModule
  ],
  providers: [provideConfig(<I18nConfig>{
    i18n: {
      resources: s4ServiceTranslations,
      chunks: s4ServiceTranslationChunksConfig,
    },
  })]
})
export class S4ServiceFeatureModule { }
"
`;

exports[`Spartacus S/4HANA Service Integration Schematics: ng-add S/4HANA Service Integration feature general setup should add the feature using the lazy loading syntax 1`] = `
"import { NgModule } from '@angular/core';
import { I18nConfig, provideConfig } from "@spartacus/core";
import { S4ServiceModule } from "@spartacus/s4-service";
import { s4ServiceTranslationChunksConfig, s4ServiceTranslations } from "@spartacus/s4-service/assets";
import { S4ServiceRootModule } from "@spartacus/s4-service/root";

@NgModule({
  declarations: [],
  imports: [
    S4ServiceRootModule,
    S4ServiceModule
  ],
  providers: [provideConfig(<I18nConfig>{
    i18n: {
      resources: s4ServiceTranslations,
      chunks: s4ServiceTranslationChunksConfig,
    },
  })]
})
export class S4ServiceFeatureModule { }
"
`;
